src/student_success_tool/analysis/pdp/dataops.py

```diff
@@ -18,6 +18,7 @@ def make_student_term_dataset(
     peak_covid_terms: set[tuple[str, str]] = constants.DEFAULT_PEAK_COVID_TERMS,
     num_terms_checkin: t.Optional[int] = None,
     key_course_subject_areas: t.Optional[list[int]] = None,
+    key_course_ids: t.Optional[list[int]] = None,
```

Comment:
Course IDs are strings, right? If so, this should be

```suggestion
    key_course_ids: t.Optional[list[str]] = None,
```